src/lib.rs

```diff
@@ -281,7 +281,9 @@ impl<C: ManageConnection + Send> Pool<C> {
                 "connection count is now: {:?}",
                 self.conn_pool.conns.total()
             );
-            self.spawn_new_future_loop();
+            if self.config.recreate_broken_connections {
+                self.spawn_new_future_loop();
```

Comment:
It seems like this has the possibility to exhaust connections. If this block doesn't run because the above is false, we could get into a situation where all connections break, we throw out all the connections, and all of the waiting tasks become stuck. The only other way to cause a new connection to be created is when a task initially asks for a connection to see if any are available. If no tasks are hitting that code path, potentially because the application reaches some concurrency limit, we will be stuck at 0 connections with lots of waiting tasks.

Reply:
I agree, I'm struggling to find a use-case for this configuration